blurring_as_a_service/cleaning_pipeline/components/smart_sampling.py

```diff
@@ -104,54 +425,140 @@ def group_files_by_date(strings):
 
     return grouped_files
 
+def get_n_random_images_per_date(grouped_images_by_date: Dict[str, List[str]], n_images_to_sample: int) -> Dict[str, List[str]]:
+    """
+    Randomly samples a specified number of images for each date.
 
-def get_10_random_images_per_date(grouped_images_by_date):
+    Parameters
+    ----------
+    grouped_images_by_date : Dict[str, List[str]]
+        A dictionary where keys are dates and values are lists of image file names from those dates.
+    n_images_to_sample : int
+        The number of images to randomly sample from each date's list.
+
+    Returns
+    -------
+    Dict[str, List[str]]
+        A dictionary where keys are dates and values are the randomly sampled image file names.
+    """
+    
     random_result = {}
 
     for key, values in grouped_images_by_date.items():
-        if len(values) >= 10:
-            random_values = random.sample(values, 10)
+        if len(values) >= n_images_to_sample:
+            random_values = random.sample(values, n_images_to_sample)
         else:
             random_values = values
         random_result[key] = random_values
 
     return random_result
 
+def connect_to_database(database_parameters_json):
```

Comment:
What is the reason to split the connect_to_database?

Reply:
The purpose was to make the database connection code reusable throughout the code instead of repeating it and to enhance readibility. In this case we only connect to the database once, so it may not be needed after all